pkgs/by-name/re/renode/common.nix

```diff
@@ -0,0 +1,118 @@
+{
+  buildDotnetModule,
+  dotnet-runtime,
+  dotnet-sdk,
+  fetchFromGitHub,
+  gcc,
+  git,
+  glibcLocales,
+  gtk3,
+  hash,
+  lib,
+  mono,
+  nugetDeps,
+  projectFile,
+  rev,
+  version,
+}:
+
+let
+  resources = fetchFromGitHub {
+    owner = "renode";
+    repo = "renode-resources";
+    rev = "d3d69f8f17ed164ee23e46f0c06844a69bf4c004";
+    hash = "sha256-wR3heL58NOQLENwCzL4lPM4KuvT/ON7dlc/KUqrlRjg=";
+  };
+  assemblyVersion =
+    s:
+    let
+      part = lib.strings.splitString "-" s;
+      result = builtins.head part;
+    in
+    result;
+
+  informationalVersion = rev: builtins.substring 0 8 rev;
+in
+buildDotnetModule rec {
+  inherit
+    dotnet-runtime
+    dotnet-sdk
+    nugetDeps
+    projectFile
+    version
+    ;
+
+  pname = "renode";
+
+  src = fetchFromGitHub {
+    owner = "renode";
+    repo = "renode";
+    rev = rev;
+    hash = hash;
+    fetchSubmodules = true;
+  };
```

Comment:
You are missing the Python Robot Framework from your work dependency. You can check the binary original recipe and likely extract it to another dependency file, allowing you to reuse it across multiple recipes.